src/Psalm/Internal/Analyzer/ClassLikeAnalyzer.php

```diff
@@ -331,6 +332,28 @@ public static function checkFullyQualifiedClassLikeName(
             return null;
         }
 
+        foreach ($class_storage->invalid_inheritors as $dependency_class_name => $_) {
+            IssueBuffer::maybeAdd(
+                new InvalidExtendClass(
+                    'Class ' . $fq_class_name . ' can be inherited by ' . implode('|', $_) . ' but parent class ' . $dependency_class_name . ' cannot',
+                    $code_location,
+                    $fq_class_name,
+                ),
+                $suppressed_issues,
+            );
+        }
+
+        if ($class_storage->active_inheritors && !in_array($fq_class_name, $class_storage->active_inheritors)) {
```

Comment:
Unsure right now on how to verify that a FQCN is a subclass of a class-string

Reply:
The easiest way is to get two Union where you have your current class in an Union as a TNamedObject and the allowed classes in the other one with other TNamedObject. Then you just have to use `UnionTypeComparator::isContainedBy` with both Union and Psalm will tell if you if the first is in the other

Reply:
Lovely, that and `TypeParser::parseTokens(TypeTokenizer::getFullyQualifiedTokens(...), ...)` got me where I needed to be.